backend/tracim_backend/lib/utils/authentication.py

```diff
@@ -20,16 +28,195 @@
+        #  The routes could also be defined through a controller (see Controller class)
+        #  even if it added-value is small.
+
+        # TODO - M.L - 2023/09/05 - Change this to proper config loading
+        self._load_settings(configurator)
+        _config = configurator.get_settings().get('pyramid_saml')
+        with open(_config.get('saml_path'), 'r') as config_file:
+            config_data = json.load(config_file)
+            self.saml_config = Saml2Config()
+            self.saml_config.load(config_data)
+            attribute_map = {
+                "EmailAddress": "user_email",
+                "UserID": "user_id",
+                "FirstName": "first_name",
+                "LastName": "last_name",
+            }
+            self.saml_config.allow_unknown_attributes = True
+            self.saml_config.attribute_map = attribute_map
+        self._metadata_response = Response(content_type='application/xml')
+        self._metadata_response.text = create_metadata_string(config=self.saml_config,
+                                                              configfile=None).decode()
+        self.saml_client = Saml2Client(config=self.saml_config)
+
+        for _, data in self.saml_client.config.vorg.items():
+            if data.common_identifier not in app_config.IDP_LIST:
+                app_config.IDP_LIST.append(data.common_identifier)
+
+        configurator.add_forbidden_view(self._idp_chooser)
+        configurator.add_route("acs", "/saml/acs", request_method="POST")
+        configurator.add_route("sso", "/saml/sso", request_method="GET")
+        configurator.add_route("metadata", "/saml/metadata", request_method="GET")
+        configurator.add_view(self._acs, route_name="acs")
+        configurator.add_view(self._sso, route_name="sso")
+        configurator.add_view(self._metadata, route_name="metadata")
+
+    def _load_settings(self, config: Configurator):
+        # TODO - M.L - 2023/09/05 - Change this to proper config loading
+        """Parse and validate SAML configuration.
+
+        Args:
+            config (pyramid.config.Configurator): The configuration of the
+                existing Pyramid application.
+        """
+
+        if 'PYRAMID_SAML_PATH' in os.environ:
+            if 'pyramid_saml' not in config.get_settings():
+                config.get_settings().update({
+                    'pyramid_saml': {}
+                })
+            config.get_settings().get('pyramid_saml').update({
+                'saml_path': os.environ.get('PYRAMID_SAML_PATH')
+            })
+
+        if 'pyramid_saml' not in config.get_settings():
+            msg = 'Missing \'pyramid_saml\' configuration in settings'
+            raise AssertionError(msg)
+        else:
+            settings = config.get_settings().get('pyramid_saml')
+
+        if 'saml_path' not in settings:
+            msg = 'Missing \'saml_path\' in settings'
+            raise AssertionError(msg)
```

Comment:
This function just tries to get the path of the JSON settings file from the config / the ENV, this could be adjusted to get the info from the `app_config` (INI file)

It basically just throws an error if the path if found nowhere